Beey.DataExchangeModel.Common/Speakers/SpeakerDtoExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Text.Json.Nodes;
+
+namespace Beey.DataExchangeModel.Common.Speakers;
+public static class SpeakerDtoExtensions
+{
+    public static string? GetString(this SpeakerDto dto, SpeakerField field)
```

Comment:
To DTO na sobě nemá žádné metody. Nehodilo by se to spíš tam? Nebo jaký má důvod to implementovat jako extensions?

Reply:
Ty chápeš SRP tak, že pokud má objekt data, tak už nemá mít logiku? Tj. že všechny objekty mají být DTO? Nebo to myslíš v tomhle konkrétním případě toho SpeakerDto?

Já pořád nechápu tu "convenience". Převedu to na jednodušší příklad - mám speciální reprezentaci pole `MyList<TKey, TValue> : IList<KeyValuePair<TKey, TValue>>` a k němu implementuju extenzi `AddItem(TKey key, TValue item)`, která zkracuje zápis pro Add, a zároveň nevkládá duplicitní prvky (to je analogické té extenzi SetAttribute pro Speakera, obecně prostě dělá něco navíc). V kódu uvidím tenhle `MyList<,>`:
1. Jak poznám, že/jestli v něm jsou nebo mají být jen unikátní prvky?
2. Jak poznám, že mám použít tu extenzi `AddItem` místo přímo `Add`, když s tím listem chci pracovat?
3. Má to být lepší, než si podědit místo IListu rovnou `IDictionary<TKey, TValue>`, který tu unikátnost zaručuje (analogie tomu, udělat atributy Speakera immutable a tu extension přesunout na něj)?